tests/src/test/scala/cats/tests/ValidatedTests.scala

```diff
@@ -2,13 +2,42 @@ package cats
 package tests
 
 import cats.data.Validated
+import cats.data.Validated.{Valid, Invalid}
 import cats.std.string._
 import cats.laws.discipline.{ApplicativeTests, SerializableTests}
 import cats.laws.discipline.arbitrary._
 import org.scalacheck.Arbitrary
+import org.scalacheck.Prop._
 
 class ValidatedTests extends CatsSuite {
-
   checkAll("Validated[String, Int]", ApplicativeTests[Validated[String,?]].applicative[Int, Int, Int])
   checkAll("Applicative[Validated[String,?]]", SerializableTests.serializable(Applicative[Validated[String,?]]))
+
+  test("ap2 combines failures in order") {
+    val plus = (_: Int) + (_: Int)
+    assert(Validated.validatedInstances[String].ap2(Invalid("1"), Invalid("2"))(Valid(plus)) == Invalid("12"))
```

Comment:
Shall we use `Applicative[Validated[String, ?]]` instead of `Validated.validatedInstances[String]`?

The latter one is subject to change if we rename the instance